compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java

```diff
@@ -88,13 +87,19 @@ void writeBBs(List<BIRBasicBlock> bbList) {
         bbList.forEach(bb -> bb.accept(this));
     }
 
-    void writeScopes(BIRAbstractInstruction instruction) {
+    void writeScopes(BIRNonTerminator instruction) {
         this.instructionOffset++;
         BirScope currentScope = instruction.scope;
 
         writeScope(currentScope);
     }
 
+    void writeScopes(BIRTerminator terminator) {
```

Comment:
writeScopes

Reply:
Yeah, that would be right since every basic block has only one terminator.